### PR TITLE
FINERACT-2062: Use 48 weeks in a year when interest rate is per month

### PR DESCRIPTION
**Pull request by wkigenyi**
## Description

Describe the changes made and why they were made.

Ignore if these details are present on the associated [Apache Fineract JIRA ticket](https://github.com/apache/fineract/pull/1284).

## Checklist

Please make sure these boxes are checked before submitting your pull request - thanks!

- [ ] Write the commit message as per https://github.com/apache/fineract/#pull-requests

- [ ] Acknowledge that we will not review PRs that are not passing the build _("green")_ - it is your responsibility to get a proposed PR to pass the build, not primarily the project's maintainers.

- [ ] Create/update unit or integration tests for verifying the changes made.

- [ ] Follow coding conventions at https://cwiki.apache.org/confluence/display/FINERACT/Coding+Conventions.

- [ ] Add required Swagger annotation and update API documentation at fineract-provider/src/main/resources/static/legacy-docs/apiLive.htm with details of any API changes

- [ ] Submission is not a "code dump".  (Large changes can be made "in repository" via a branch.  Ask on the developer mailing list for guidance, if required.)

FYI our guidelines for code reviews are at https://cwiki.apache.org/confluence/display/FINERACT/Code+Review+Guide.

### CODE REVIEW

**Comment by adamsaghy** [comment]
Please kindly see the discussion under the JIRA story: https://issues.apache.org/jira/browse/FINERACT-2062

Also please make sure you are following the proper commit message requirements:
"FINERACT-<JIRA_STORY_NUMBER>: <a short sentence about the change>"
Example:  
FINERACT-2062: Use 48 weeks in a year if interest rate is monthly

**Review by adamsaghy — CHANGES_REQUESTED** [review]
Kindly check my comments!

**Comment by wkigenyi** [comment]
Thanks for the guidance @adamsaghy. I am following the conversation jira.

File: fineract-provider/src/main/java/org/apache/fineract/portfolio/loanaccount/loanschedule/domain/DefaultPaymentPeriodsInOneYearCalculator.java [hunk 1]
```diff
@@ -32,15 +32,15 @@
...
 
     @Override
-    public Integer calculate(final PeriodFrequencyType repaymentFrequencyType) {
+    public Integer calculate(final PeriodFrequencyType repaymentFrequencyType,final PeriodFrequencyType interestRateFrequencyMethod) {
 
         Integer paymentPeriodsInOneYear = 0;
         switch (repaymentFrequencyType) {
             case DAYS:
-                paymentPeriodsInOneYear = 365;
+                paymentPeriodsInOneYear = interestRateFrequencyMethod.isMonthly() ? Integer.valueOf(360) : Integer.valueOf(365);
```
Comment (adamsaghy) [inline fineract-provider/src/main/java/org/apache/fineract/portfolio/loanaccount/loanschedule/domain/DefaultPaymentPeriodsInOneYearCalculator.java:40]: This looks weird to me. If why would it be 360 if it is monthly?

File: fineract-provider/src/main/java/org/apache/fineract/portfolio/loanaccount/loanschedule/domain/DefaultPaymentPeriodsInOneYearCalculator.java [hunk 2]
```diff
@@ -32,15 +32,15 @@
...
             break;
             case WEEKS:
-                paymentPeriodsInOneYear = 52;
+                paymentPeriodsInOneYear = interestRateFrequencyMethod.isMonthly() ? Integer.valueOf(48) : Integer.valueOf(52);
```
Comment (adamsaghy) [inline fineract-provider/src/main/java/org/apache/fineract/portfolio/loanaccount/loanschedule/domain/DefaultPaymentPeriodsInOneYearCalculator.java:43]: This looks weird to me. If why would it be 48 if it is monthly?

**Review by adamsaghy — CHANGES_REQUESTED** [review]
Kindly check my review!

File: fineract-provider/src/main/java/org/apache/fineract/portfolio/loanaccount/loanschedule/domain/DefaultPaymentPeriodsInOneYearCalculator.java (hunk 2 above)
Comment (wkigenyi) [reply fineract-provider/src/main/java/org/apache/fineract/portfolio/loanaccount/loanschedule/domain/DefaultPaymentPeriodsInOneYearCalculator.java:43]: If the interest frequency is monthly, it looks more reasonable to assume there are four weeks in a month, so with 4 weeks in a month and 12 months in a year we shall have 4 x 12 = 48 weeks in a year.

File: fineract-provider/src/main/java/org/apache/fineract/portfolio/loanaccount/loanschedule/domain/DefaultPaymentPeriodsInOneYearCalculator.java (hunk 1 above)
Comment (wkigenyi) [reply fineract-provider/src/main/java/org/apache/fineract/portfolio/loanaccount/loanschedule/domain/DefaultPaymentPeriodsInOneYearCalculator.java:40]: With monthly interest interest and daily repayments, it looks more reasonable to assume that there are exactly 360 days, a number divisible by the 12 months in the year brings our more explainable installments.

File: fineract-provider/src/main/java/org/apache/fineract/portfolio/loanaccount/loanschedule/domain/DefaultPaymentPeriodsInOneYearCalculator.java (hunk 2 above)
Comment (adamsaghy) [reply fineract-provider/src/main/java/org/apache/fineract/portfolio/loanaccount/loanschedule/domain/DefaultPaymentPeriodsInOneYearCalculator.java:43]: I dont think thats right... monthly interest calculation cannot be 4 weeks... @bharathc27 Thoughts?

**Pull request closed without merging**